### DISCUSSION TITLE
16:9 layout for the card

### DISCUSSION THREAD

**Discussion by LucianTothezan** [slipx06/sunsynk-power-flow-card#542] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

the card layout is a square so using it as a single card in Panel layout on a tablet would not display the card on the whole area of the 16:9 display.

### Expected behaviour

I would like to use this card as Panel on a wall mount tablet so the card should cover the whole display area. Therefore a 16:9 and/or 9:16 layout would be great.

### Mode

Both

### Context / Reason

wallmount 16:9 tablet with this card as Panel display.

**Comment by slipx06** [comment]
The card is rendered as an SVG element with a viewbox that is set to preserve aspect ratio . At this stage there is no quick fix. You can try adjust the card_height: , card_width: and panel: attributes. You can also experiment with different dashboard layouts i.e. grid layout etc

**Comment by LucianTothezan** [comment]
Thank You for the reply. I really really appreciate this card. I have
already tried what you suggested but unfortunately the result remains the
same: if panel mode I can modify the height but it will auto adjust also
the width. In grid mode also, it will modify the width but also
automatically will keep the aspect ratio. It would really need an 16:9 SVG
that way if we modify the width or the height, it would preserve the native
wide format of the SVG. Unfortunately I do not have the proper knowledge to
create such an element by myself....

Best regards,

Lucian Tothezan

lun., 28 oct. 2024, 19:20 slipx06 ***@***.***> a scris:

**Comment by slipx06** [comment]
Its a lot of work as all the elements need to be re-positioned.  Moving this to an idea and maybe someone will pick it up

**Reply by LucianTothezan** [reply]
i would like to try to modify it but i am a total noob. i think i get the jist of it but i do not know how You then "pack" all the source code into one .js file. Could You please give me a pointer into the right direction? thank You in advance. Best regards.

L.E. neverind, i found the way to generate the .js file.

**Comment by LucianTothezan** [comment]
after a few hours of "reading" the code, i have a result which still can be improved but it is the goal i was setting.
![WhatsApp Image 2024-11-03 at 08 07 08](https://github.com/user-attachments/assets/0f528677-050c-49d1-863c-57d4a23f2d21)

**Reply by slipx06** [reply]
Well done for figuring that out. Maybe you can share your changes in case other users want to try this

**Comment by LucianTothezan** [comment]
Thank You for the kind words. Yes I could just that I have modified only
the full card and only the entities that I have now on display. It's only
half work done.

dum., 3 nov. 2024, 12:11 slipx06 ***@***.***> a scris:
